### PR TITLE
Concurrency

### PR DESCRIPTION
**Pull request by IvanovDmytroA**

### CODE REVIEW

**Review by yurchenkovr — APPROVED** [review]

File: cmd/main.go [hunk 1]
```diff
@@ -1,7 +1,14 @@
...
-import "github.com/IvanovDmytroA/lets-go-chat/internal/server"
+import (
+	"sync"
+
+	"github.com/IvanovDmytroA/lets-go-chat/internal/server"
+)
 
 func main() {
-	server.Start()
+	once := sync.Once{}
```
[the remark this answers is not in the document]
Comment (IvanovDmytroA) [reply cmd/main.go]: Removed unnecessary code

File: internal/handler/websocket/client.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+		return nil, err
+	}
+	claims, ok := vt.Claims.(jwt.MapClaims)
+	if ok && vt.Valid {
+		accessUuid, ok := claims["access_uuid"].(string)
+		if !ok {
+			return nil, err
+		}
+		userId := claims["user_id"].(string)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (IvanovDmytroA) [reply internal/handler/websocket/client.go]: Added bool check

File: internal/handler/websocket/client.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	if err != nil {
+		errMsg := "Invalid token: "
+		log.Printf(errMsg+"%v\n", err)
+		return echo.NewHTTPError(http.StatusBadRequest, errMsg+err.Error())
+	}
+
+	return connect(c, *accessDetails)
+}
+
+func connect(c echo.Context, ad AccessDetails) error {
```
[the remark this answers is not in the document]
Comment (IvanovDmytroA) [reply internal/handler/websocket/client.go]: Done

File: internal/handler/websocket/client.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	redClient, _ := c.Get("redis").(*redis.Client)
```
[the remark this answers is not in the document]
Comment (IvanovDmytroA) [reply internal/handler/websocket/client.go]: Added ok variable and verification

**Pull request merged**